src/plstring.c

```diff
@@ -0,0 +1,540 @@
+#include <nitro.h>
+#include <string.h>
+
+#include "struct_decls/struct_plstring_decl.h"
+
+#include "heap.h"
+#include "plstring.h"
+
+// Important character constants
+#define CR                      0xe000
+#define TERMINATOR              0xffff
+#define IS_COMPRESSED_MASK      0xf100
+#define COMPRESSION_MASK        0x01ff
+#define COMPRESSED_TERMINATOR   0x01ff
```

Comment:
These might be too generic of names, think of a prefix. (However, these are also "static" defines, so they won't conflict elsewhere in source code)

Also, what is the difference between `IS_COMPRESSED_MASK` and `COMPRESSION_MASK`?

Reply:
I've updated these a bit:

```c
#define COMPRESSED_LEADER       0xf100
#define COMPRESSED_CODE_MASK    0x01ff
#define COMPRESSED_CODE_BITS    9
#define COMPRESSED_EOM          0x01ff
#define CHARCODE_VALID_BITS     15
```